frontend/components/AuthorTable.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useCallback } from "react";
+import { Input, Button, DatePicker } from "antd";
+import { useMutation, useQuery } from "@apollo/client";
+import { AuthorsData, Author } from "@/types/author";
+import { ADD_AUTHOR, GET_AUTHORS, UPDATE_AUTHOR } from "@/utils/graphql";
+import PaginatedTable from "./PaginatedTable";
+import AuthorForm from "./AuthorForm";
+import moment from "moment";
+
+const { RangePicker } = DatePicker;
+
+const AuthorTable: React.FC = () => {
+  const { refetch } = useQuery<AuthorsData>(GET_AUTHORS);
+  const [addAuthor, { data }] = useMutation(ADD_AUTHOR);
+  const [updateAuthor, { data: updatedData }] = useMutation(UPDATE_AUTHOR);
```

Comment:
When we update the AUhtors, we already get back the authors from the mutation, how can we use that itself, instead of doing a refetch again?

Reply:
No we are doing a refetch. There's a useCallback below that triggers the refetch function.